StabilityPackage.wl

```diff
@@ -211,18 +357,112 @@ Reap[
 		ktarget=(S/.kvar->0)[[1,1]];
 		kl=buildkGrid[ktarget,nstep];
 		Do[
-			evals=Sort[Im[evscale[kl[[kx]],S,kvar]],Greater];
-			pot=potential[data,ea]/.kvar->kl[[kx]];
-			Sow[{data["radius"][[rx]],kl[[kx]],evals[[1]],pot}]; 
+		ea2=getEquations[singleRadiusData,testE,hi,kx];
+		S2=stabilityMatrix[data,ea];
+			gdout=GDValue[S2]; (*this isn't updating k correctly*)
+			Sow[{data["radius"][[rx]],kl[[kx]],gdout}]; 
 		,{kx,1,Length[kl]}] (*close do over ktargets*)
 	,{rx,rstr,rend}] (*close do over r*)
 ][[2,1]];
-Return[evout] (*Close reap over r*)
-]; (*close module*)
+Return[out] (*Close reap over r*)
+		
+]
+
+getIntialGuess[data_,species_]:=Module[{ei,aei,ag,aag,\[Beta]g,a\[Beta]g,a\[Chi]g,\[Chi]g,g0,datasr,foc1234,afoc1234,xfoc1234,xag,xei},
+datasr=SelectSingleRadius[data,1];
+
+foc1234[x_,y_,z_,E_]:= ((3c^3)/(4 Pi h (1/2 (data["freq"][[E+1]]+data["freq"][[E]])) (data["freq"][[E+1]]^3-data["freq"][[E]]^3)) )(datasr["lotsodo"][[1,E,1]] 
```

Comment:
would be better to combine these functions with a "species index" parameter and a "weight" parameter to prevent repeated code

Reply:
You mean combine the three "foc1234"s right?